### PR TITLE
Implement deleteT

### PR DESCRIPTION
**Pull request by timleow**
### Commands implemented:

`deleteT`, to delete tasks from a certain project.

Multiple tasks can be deleted at once.

### Command format:
`deleteT pr/PROJECT_INDEX t/TASK_INDEXES e.g. deleteT pr/1 t/1 2 3`

#### Before execution of `deleteT`:
image

#### After execution:
image

### Other notes:
- Minor implementation note: Indexes are sorted in descending order, so that deleting them in that order maintains the indexes of items that are before the current item to delete.

        taskIndexes.sort((a, b) -> b.getZeroBased() - a.getZeroBased());

- Also changed task deadline formats:
`addT n/TASK_NAME d/DEADLINE`~yyyy-MM-dd HHmm~ dd-mm-yyyy HHmm

- Ui update: project cards will not display the `tasks` section if the project contains no tasks. Project 1 does not have any tasks in the below screenshot, while project 2 has one task.
- 
image

- Also ensured that no duplicate prefixes are allowed for projects, tasks in `markT`, `unmarkT`, `deleteT`

### CODE REVIEW

**Review by Chandan8186 — APPROVED** [review]
LGTM!

**Review by rayyan35p — APPROVED** [review]
Just some very minor comments, other than that LGTM!

File: src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public CommandResult execute(Model model) throws CommandException {
+        List<Project> lastShownProjectList = model.getFilteredProjectList();
+
+        if (lastShownProjectList.size() == 0) {
+            throw new CommandException(Messages.MESSAGE_NO_PROJECT_TO_DELETE_TASK);
+        } else if (projectIndex.getZeroBased() >= lastShownProjectList.size()
+            // Check if project index is valid first
+            || projectIndex.getZeroBased() < 0) {
+            logger.warning("Invalid project index: " + projectIndex.getOneBased());
+            throw new CommandException(Messages.MESSAGE_INVALID_PROJECT_DISPLAYED_INDEX);
```
[the remark this answers is not in the document]
Comment (timleow) [reply src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java:63]: That's possible, but note that this code is heavily similar to `markT` and `unmarkT`- I'll relook the error handling for them and make changes in a future PR if necessary.

File: src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        }
+
+        Project targetProject = lastShownProjectList.get(projectIndex.getZeroBased());
+        TaskList lastShownTaskList = targetProject.getTasks();
+
+        // Delete all tasks from targetProject
```
[the remark this answers is not in the document]
Comment (timleow) [reply src/main/java/seedu/address/logic/commands/DeleteTaskCommand.java:69]: Will update and push in a future PR, thanks for the catch!

**Pull request merged**